botocore/config.py

```diff
@@ -206,6 +206,21 @@ class Config:
         creating new connections if set to True.
 
         Defaults to False.
+
+    :type request_min_compression_size_bytes: int
+    :param request_min_compression_bytes: The minimum size in bytes that a
+    request body should be to trigger compression. A request is compressed
+    regardless of the size of the body if the operation has streaming input,
+    but if the operation also contains the `requiresLength` trait, the request
+    is not compressed.
```

Comment:
> A request is compressed regardless of the size of the body if the operation has streaming input,
    but if the operation also contains the `requiresLength` trait, the request
    is not compressed.

We have this written in a few places but it seems incorrect. We don't compress streaming input regardless of body size. We will compress all streaming input that doesn't require length.

Reply:
Per discussion offline updated wording in the latest commit.